### ISSUE TITLE
Image with Filter "[/FlateDecode/JPXDecode]" not extracted

### ISSUE THREAD

**Issue by AlexMatiash** [pymupdf/PyMuPDF#2087]
## Describe the bug (mandatory)
I'm trying to iterate over all images in the document, but for one filter combination it fails.
PyMuPDF works correctly for all filter combinations except `Filter [/FlateDecode/JPXDecode]`. Pdfs with such image filters are correctly read by pdf readers and other python pdf libs, but PyMuPDF fails to extract image and get correct filters by `xref_get_key(xref, "/Filters")`.

these:
`<</ColorSpace/DeviceRGB/BitsPerComponent 8/Width 1672/Length 1964389/Height 1124/Name/im1/Subtype/Image/Type/XObject/Filter/JPXDecode>>`

and

`<</ID 26 0 R/Type/XObject/Length 476/Filter[/FlateDecode/DCTDecode]/Subtype/Image/BitsPerComponent 8/Width 126/Height 81/ColorSpace/DeviceRGB>>`

are **ok**

this
`<</ColorSpace/DeviceGray/BitsPerComponent 8/Width 67/Length 1958/Height 68/Name/im2/Subtype/Image/Type/XObject/Filter[/FlateDecode/JPXDecode]>>`

**fails**

**However**, `document.xref_stream(xref)` correctly decompresses the stream and output is valid jpeg2000 stream.

## To Reproduce (mandatory)
```
    document = fitz.Document(srcFileName)
    allXrefsLength = document.xref_length()
    for xref in range(1, allXrefsLength):
        if document.xref_get_key(xref, "Subtype")[1] != "/Image":
            continue

        imgDict = document.extract_image(xref)
        if not imgDict:
            tmpFilters = document.xref_get_key(xref, "/Filters")
            print("subtype of xref {0} is /Image, but pymupdf can not extract it as image. filters: {1}".format(xref, tmpFilters))
```
output for images with such filters:
`subtype of xref 76 is /Image, but pymupdf can not extract it as image. filters: ('null', 'null')`

## Your configuration (mandatory)
Windows 10 x64, python 3.10, pymupdf 1.21, installed by pip install pymupdf

**Comment by JorjMcKie** [comment]
Duplicate of #1995: the maximum supported values for image width and height currently are 2**16 = 65,536.
See that issue for details.

**Comment by AlexMatiash** [comment]
> Duplicate of #1995: the maximum supported values for image width and height currently are 2**16 = 65,536. See that issue for details.

It happens with all image sizes, in my example it is 67 x 68 px. Only the filter combination matters.

**Comment by JorjMcKie** [comment]
You have a typo in extracting the filter names: `/Filters` does not exist, only `/Filter`.

**Comment by JorjMcKie** [comment]
Please provide a reproducer file.

**Comment by JorjMcKie** [comment]
Thanks for the file.
Fixed it.
Will be published with next version.

**Comment by AlexMatiash** [comment]
Thank you!

**Comment by julian-smith-artifex-com** [comment]
Fixed in PyMuPDF-1.21.1.
